.github/workflows/release.yaml

```diff
@@ -1,53 +1,110 @@
 name: Release Packages and Docker images
 on:
-  push:
-    tags:
-      - '[0-9]+.[0-9]+.[0-9]+'
+  release:
+    types: [published]
 
 # Checks if any concurrent jobs is running for release CI and eventually cancel it.
 concurrency:
   group: ci-release
   cancel-in-progress: true  
 
 jobs:
+  release-settings:
+    runs-on: ubuntu-latest
+    outputs:
+      is_latest: ${{ steps.get_settings.outputs.is_latest }} 
+      bucket_suffix: ${{ steps.get_settings.outputs.bucket_suffix }}
+      should_publish: ${{ steps.get_settings.outputs.should_publish }}
+    steps:
+      - name: Get latest release
+        uses: rez0n/actions-github-release@v2.0
+        id: latest_release
+        env:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          repository: ${{ github.repository }}
+          type: "stable"
+
+      - name: Get settings for this release
+        id: get_settings
+        shell: python
+        run: |
+          import os
+          import re
+
+          semver_no_meta = '''^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?$'''
+          tag_name = '${{ github.event.release.tag_name }}'
+
+          should_publish = re.match(semver_no_meta, tag_name) is not None
+
+          is_prerelease = '-' in tag_name
+
+          # Safeguard: you need to both set "latest" in GH and not have suffixes to overwrite latest
+          is_latest = '${{ steps.latest_release.outputs.release }}' == tag_name and not is_prerelease
+
+          bucket_suffix = '-dev' if is_prerelease else ''
+
+          with open(os.environ['GITHUB_OUTPUT'], 'a') as ofp:
+            print(f'is_latest={is_latest}'.lower(), file=ofp)
+            print(f'should_publish={should_publish}'.lower(), file=ofp)
+            print(f'bucket_suffix={bucket_suffix}', file=ofp)
+
   build-packages:
+    needs: [release-settings]
+    if: ${{ needs.release-settings.outputs.should_publish == 'true' }}
```

Comment:
I'd still build artifacts even if we are not going to publish them (they can still be accessed as github action run artifact).

Reply:
So, should_publish means that this is a release with a tag name we understand (e.g. `0.1.2-rc1` or `3.4.5`) and will be false for anything that is not that (a release tagged `foo`). I'm not sure what should we do in those cases so I picked nothing. Do you have suggestions? A better variable name?

Reply:
fail the pipeline upon a `git tag`? would that make sense? honest question

Reply:
IMHO yep since that's a bug for us (ie: we don't want to support non semversioned tags i think).

Reply:
If `should_publish` is false the whole workflow won't do nothing, right? At least if we fail immediately, we will have the nice red cross telling us something went wrong.